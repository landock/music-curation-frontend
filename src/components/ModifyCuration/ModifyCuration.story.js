import React from 'react';

import { MemoryRouter as Router } from 'react-router';
import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { linkTo } from '@storybook/addon-links';

import ModifyCuration from '.';

storiesOf('ModifyCuration', module)
  .addDecorator(story => {
    return <Router>{story()}</Router>;
  })
  .add('default', () => <ModifyCuration />);
