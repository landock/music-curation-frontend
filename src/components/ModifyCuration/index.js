import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Menu, Icon } from 'semantic-ui-react';

class ModifyCuration extends PureComponent {
  render() {
    return (
      <Menu text vertical>
        <Menu.Item link name="Edit">
          <Icon name="write" />
          <Link to={`editCuration/${this.props.curationId}`}>Edit</Link>
        </Menu.Item>
        <Menu.Item link name="Delete">
          <Icon name="trash" />
          Delete
        </Menu.Item>
        <Menu.Item link name="Duplicate">
          <Icon name="plus" />
          Duplicate
        </Menu.Item>
      </Menu>
    );
  }
}

ModifyCuration.propTypes = {};

export default ModifyCuration;
