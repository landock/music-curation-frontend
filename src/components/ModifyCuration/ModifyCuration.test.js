import React from 'react';
import { MemoryRouter as Router } from 'react-router-dom';
import { render } from 'react-testing-library';
import renderer from 'react-test-renderer';

import ModifyCuration from '.';

it('ModifyCuration: default', () => {
  const component = renderer.create(
    <Router>
      <ModifyCuration />
    </Router>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
