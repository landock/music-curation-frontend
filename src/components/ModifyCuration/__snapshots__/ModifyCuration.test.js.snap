// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModifyCuration: default 1`] = `
<div
  className="ui text vertical menu"
>
  <div
    className="link item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="write icon"
    />
    <a
      href="editCuration/undefined"
      onClick={[Function]}
      style={
        Object {
          "color": "rgba(0,0,0,.6)",
        }
      }
    >
      Edit
    </a>
  </div>
  <div
    className="link item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="trash icon"
    />
    Delete
  </div>
  <div
    className="link item"
    onClick={[Function]}
  >
    <i
      aria-hidden="true"
      className="plus icon"
    />
    Duplicate
  </div>
</div>
`;
